contrib/golang/common/go/api/type.go

```diff
@@ -317,8 +317,10 @@ type ConnectionEvent int
 const (
 	RemoteClose      ConnectionEvent = 0
 	LocalClose       ConnectionEvent = 1
-	Connected        ConnectionEvent = 2
-	ConnectedZeroRtt ConnectionEvent = 3
+	RemoteReset      ConnectionEvent = 2
+	LocalReset       ConnectionEvent = 3
+	Connected        ConnectionEvent = 4
+	ConnectedZeroRtt ConnectionEvent = 5
```

Comment:
I'm worried about the impact of extending this enum on proprietary extensions. If someone misses this change and does not update their filter to handle or check for the new values they can end up diagnosing hard problems.

Would it make more sense to make the type of connection closure available through the Network::Connection interface, once the connection was closed? I think it will also significantly shrink the size of this PR. 

Conceptually very few users will care about the manner in which connection was closed, and just need a closed event. So it make more sense to me to make this property of the close event to just be accessible through the Connection interface. This option was discussed in the #2703 as well.

Reply:
Yeah, this is a good point, and that is why I disabled it by default.

This is more for the notification for the downstream or upstream. The only downside is we may need to add more parameters like close type to notify the upstream/downstream when the other is reseted to the callback methods like the onDownstreamEvent() for upstream when the connection object is not exposed to the other.

Reply:
Done for this idea. Thanks!

Reply:
Hi @yanavlasov, PTAL, thanks!